### ISSUE TITLE
Optimize images with SMask

### ISSUE THREAD

**Issue by benbro** [ocrmypdf/OCRmyPDF#1073]
**Is your feature request related to a problem? Please describe.**
Some tools create PDFs with huge images. [optimize.py](https://github.com/ocrmypdf/OCRmyPDF/blob/c1f99c0bd54303d3bb3f9df4daa9a586e08e3ced/src/ocrmypdf/optimize.py#L261) currently doesn't optimize images with SMask. Is there a way to improve it?

**Describe the solution you'd like**
Downsample/optimize images with SMask.

**Describe alternatives you've considered**
ghostscript can optimize and downsample an image with SMask:
```gs -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dPDFSETTINGS=/screen -sOutputFile=optimized.pdf test.pdf```

**Example file**
This pdf created with an old version of PowerPoint and has an image with SMask:
[test.pdf](https://github.com/ocrmypdf/OCRmyPDF/files/10822989/test.pdf)

**Additional context**

**Comment by jbarlow83** [comment]
For the file in question, the image is inside a Form XObject which we don't currently attempt to optimize, so it's excluded for that reason. The SMask is already compressed with Flate to a high ratio and likely can't be improved.

SMask is a bit of special case. We cannot use pngquant on SMask, because pngquant tries to find a palette and PDF doesn't accept palettes on SMask. That's the main reason they are skipped.

In the rare case of lossy SMask we should probably leave them alone.

We could also apply Flate to any SMask that is not compressed. (We can that by saving the file PDF with recompress_flate in pikepdf rather than the normal image optimize, just by changing `get_pdf_save_settings`, although the savings achieved won't be logged as usual.) That's fairly easy to do.

We could potentially export the image+SMask as RGBA to pngquant, and then split the <8-bit RGBA palette image into a <8-bit palette image and 8-bit SMask. This gets complicated and needs to be checked in both the premultiplied (SMask with Matte set) and straight cases, but might lead to bigger savings on both images combined.

**Comment by benbro** [comment]
I'm asking about option to downsample SMask because in the attached pdf the it is large for no apparent reason. If I'm not mistaken, large SMask will result with large memory usage and longer rendering time in pdf viewers.
I'm less interested in reducing the pdf size for this specific case.

**Comment by jbarlow83** [comment]
Yes, an SMask that large will take excessive resources for rendering and will end up downsampled by the PDF viewer anyway. We'll have to see about opt-in downsampling in a future release.
